lib/jnpr/junos/facts/get_software_information.py

```diff
@@ -17,6 +17,12 @@ def _get_software_information(device):
             except Exception:
                 pass
         try:
+            # JDM for Junos Node Slicing
+            return device.rpc.cli("show version all-servers", format='xml',
```

Comment:
No, at the time this runs, there aren't really any facts available that we could use for an if block rather than a try/catch block.

I will convert to using the suggested RPC since it does seem to work in this case. I hadn't used the RPC because `| display xml rpc` didn't show it.
```
root@jdm> show version all-servers | display xml rpc 
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/17.4I0/junos">
    <rpc>
        <get-software-information>
        </get-software-information>
    </rpc>
    <cli>
        <banner></banner>
    </cli>
</rpc-reply>
```